Scripts/AI/MotionController/MotionControllerAIController.cs

```diff
@@ -0,0 +1,35 @@
+﻿using com.ootii.Actors.AnimationControllers;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace NeoFPS.AI.ootii
+{
+    public class MotionControllerAIController : BasicAIController
+    {
+        private MotionController m_MotionController;
+
+        protected override void Awake()
+        {
+            base.Awake();
+            m_MotionController = GetComponent<MotionController>();
+            Debug.Assert(m_MotionController != null, gameObject + " has a MotionControllerAIController but no motionController component.");
+        }
+
+        private void OnEnable()
```

Comment:
If you intend people to override this then best to make any Unity methods like this protected virtual too